buildkite/data_source_team.go

```diff
@@ -2,84 +2,118 @@ package buildkite
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This datasource allows you to get a team from Buildkite.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the team.",
+				Required:    true,
 			},
-			"slug": {
-				Required: true,
-				Type:     schema.TypeString,
+			"uuid": schema.StringAttribute{
+				Description: "The UUID of the team.",
+				Computed:    true,
 			},
-			"uuid": {
-				Computed: true,
-				Type:     schema.TypeString,
+			"slug": schema.StringAttribute{
+				Description: "The slug of the team.",
+				Computed:    true,
 			},
-			"name": {
-				Computed: true,
-				Type:     schema.TypeString,
+			"name": schema.StringAttribute{
+				Description: "The name of the team.",
+				Computed:    true,
 			},
-			"privacy": {
-				Computed: true,
-				Type:     schema.TypeString,
+			"privacy": schema.StringAttribute{
+				Description: "The privacy of the team.",
+				Computed:    true,
 			},
-			"description": {
-				Computed: true,
-				Type:     schema.TypeString,
+			"description": schema.StringAttribute{
+				Description: "The description of the team.",
+				Computed:    true,
 			},
-			"default_team": {
-				Computed: true,
-				Type:     schema.TypeBool,
+			"default_team": schema.BoolAttribute{
+				Description: "Whether the team is the default team.",
+				Computed:    true,
 			},
-			"default_member_role": {
-				Computed: true,
-				Type:     schema.TypeString,
+			"default_member_role": schema.StringAttribute{
+				Description: "The default member role of the team.",
+				Computed:    true,
 			},
-			"members_can_create_pipelines": {
-				Computed: true,
-				Type:     schema.TypeBool,
+			"members_can_create_pipelines": schema.BoolAttribute{
+				Description: "Whether members can create pipelines.",
+				Computed:    true,
 			},
 		},
 	}
 }
 
-// ReadTeam retrieves a Buildkite team
-func dataSourceTeamRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	// Warning or errors can be collected in a slice type
-	var diags diag.Diagnostics
+func (t *teamDatasource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state teamDatasourceModel
 
-	client := m.(*Client)
-	orgTeamSlug := fmt.Sprintf("%s/%s", client.organization, d.Get("slug").(string))
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
-	response, err := getTeam(client.genqlient, orgTeamSlug)
+	res, err := getTeam(t.client.genqlient, state.ID.ValueString())
 
 	if err != nil {
-		return diag.FromErr(err)
+		resp.Diagnostics.AddError(
+			"Unable to get team",
+			fmt.Sprintf("Error getting team: %s", err.Error()),
+		)
+		return
 	}
 
-	if response.Team.Id == "" {
-		return diag.FromErr(errors.New(fmt.Sprintf("Team not found: '%s'", d.Get("slug"))))
+	if converted, ok := res.GetNode().(*getTeamNodeTeam); ok {
```

Comment:
This shouldn't happen anyway, but if its not `ok`, we should error out and not set the state. I think that's the safest approach